src/app/static/src/app/store.ts

```diff
@@ -0,0 +1,37 @@
+import Vue from "vue"
+import Vuex from "vuex"
+import {MutationPayload, Store, StoreOptions} from "vuex";
+
+export interface Region {
+    name: string
+}
+
+export interface Project {
+    name: string
+    regions: Region[]
+}
+
+export interface RootState {
+    projects: Project[]
+    currentProject: Project | null
+    currentRegion: Region | null
```

Comment:
I suppose another way to structure this would be to have currentRegion be a property of Project rather than at the top level so you'd do currentProject.currentRegion. That way there shouldn't be any risk of the current region coming adrift from the current project, and the project would automatically persist its last selected region

Reply:
yes that's much better, thanks